.github/workflows/swan_pr_workflow.yml

```diff
@@ -0,0 +1,146 @@
+# GitHub action file for SwanHub SDK
+name: SwanHub SDK Application Workflow
+
+# Triggers the workflow on push or pull request
+on:
+  push: # Triggered on a push to any branch ( do not change )
+  schedule:
+    # Run workflow at 4 AM EST (9 AM UTC)
+    - cron: '0 9 * * *'
+    # Run workflow at 7 PM EST (12 AM UTC next day)
+    - cron: '0 0 * * *'
+  release:
+    types: [ created ] # Triggers the workflow when a release is created.
+
+# Defines the jobs that the workflow will execute
+jobs:
+  Lint:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v4
+      - name: Install black
+        run: |
+          pip install black
+      - name: Lint with Black
+        run: |
+          black --check ./**/*.py
+  Test:
+    needs: Lint
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v4
+      - name: Install pytest
+        run: |
+          pip install -r requirements.txt
+      - name: Run tests
+        run: pytest
+
+  Build:
+    needs: Test
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v4
+      - name: Build the project
+        run: |
+          python setup.py sdist bdist_wheel
```

Comment:
do we have cleanup in place for the /dist folder, because otherwise the upload will upload all files in the /dist folder

Reply:
Also if we run this action on every single push then we would build the project every single time, maybe make a different GitHub action for deployment and another for building the image?

Reply:
I made changes that, build will be on merge to dev/main only

Reply:
you are so good DJ, interestingly good